spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -677,6 +677,12 @@ class DistriOptimizer[T: ClassTag] (
 
   private var models: RDD[DistriOptimizer.Cache[T]] = null
 
+  import DistriOptimizer._
+  if (dataset.originRDD != null) {
+    logger.info("Loading and caching data ...")
+    dataset.originRDD().cache().count()
```

Comment:
I don't agree optimizer do anything related with manipulate data, which should be delegate to dataset. The name of dataset is CachedDataset, so it cache the data for performance. If we have a new dataset named NoCacheDataSet to handle extra large data, will you cache it again?